File: E-mart_.Net/Backend/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace E_mart.Models;

[Table("products")]
[Index("Categoryid", Name = "FK1krrsjgcawsfg8k8u4hm5gi8q")]
[Index("Subcategoryid", Name = "FKi5prgvmxtqxb5fji9wc2qivr2")]
public partial class Product
{
    [Key]
    [Column("productid")]
    public int Productid { get; set; }

    [Column("brand")]
    [StringLength(255)]
    public string? Brand { get; set; }

    [Column("created_at")]
    [MaxLength(6)]
    public DateTime? CreatedAt { get; set; }

    [Column("description")]
    [StringLength(255)]
    public string? Description { get; set; }

    [Column("description_en")]
    [StringLength(255)]
    public string? DescriptionEn { get; set; }

    [Column("description_fr")]
    [StringLength(255)]
    public string? DescriptionFr { get; set; }

    [Column("discount")]
    public double Discount { get; set; }

    [Column("image")]
    [StringLength(255)]
    public string? Image { get; set; }

    [Column("isdeal", TypeName = "bit(1)")]
    public ulong Isdeal { get; set; }

    [Column("price")]
    public double Price { get; set; }

    [Column("productname")]
    [StringLength(255)]
    public string Productname { get; set; } = null!;

    [Column("productname_en")]
    [StringLength(255)]
    public string? ProductnameEn { get; set; }

    [Column("productname_fr")]
    [StringLength(255)]
    public string? ProductnameFr { get; set; }

    [Column("rating")]
    public double Rating { get; set; }

    [Column("stocks")]
    public int Stocks { get; set; }

    [Column("updated_at")]
    [MaxLength(6)]
    public DateTime? UpdatedAt { get; set; }

    [Column("categoryid")]
    public int Categoryid { get; set; }

    [Column("subcategoryid")]
    public int? Subcategoryid { get; set; }

    [InverseProperty("Product")]
    public virtual ICollection<Cartitem> Cartitems { get; set; } = new List<Cartitem>();

    [InverseProperty("Product")]
    public virtual ICollection<Cart> Carts { get; set; } = new List<Cart>();

    [ForeignKey("Categoryid")]
    [InverseProperty("Products")]
    public virtual Category Category { get; set; } = null!;

    [ForeignKey("Subcategoryid")]
    [InverseProperty("Products")]
    public virtual Subcategory? Subcategory { get; set; }
}

File: E-mart_java/Backend/src/main/java/com/example/Models/Product.java
package com.example.Models;

import java.util.Locale;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;


import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "productid", nullable = false, unique = true)
    private int id;

    @Column(nullable = false)
    private String productname;

    private String description;
    private double rating;
    private int stocks;
    private String brand;
    
    private String image;

    public String getImage() {
		return image;
	}
	public void setImage(String image) {
		this.image = image;
	}



	@ManyToOne
    @JoinColumn(name = "categoryid", nullable = false)
    private Category category;

    @ManyToOne
    @JoinColumn(name = "subcategoryid")
    private Subcategory subcategory;

    private double discount;
    private double price;
    private boolean isdeal = false;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    
//	public int getProductid() {
//		return productid;
//	}
//	public void setProductid(int productid) {
//		this.productid = productid;
//	}
//    
    
	public String getProductname() {
		return productname;
	}
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public void setProductname(String productname) {
		this.productname = productname;
	}
	public String getDescription() {
		return description;
	}
	public void setDescription(String description) {
		this.description = description;
	}
	public double getRating() {
		return rating;
	}
	public void setRating(double rating) {
		this.rating = rating;
	}
	public int getStocks() {
		return stocks;
	}
	public void setStocks(int stocks) {
		this.stocks = stocks;
	}
	public String getBrand() {
		return brand;
	}
	public void setBrand(String brand) {
		this.brand = brand;
	}
	public Category getCategory() {
		return category;
	}
	public void setCategory(Category category) {
		this.category = category;
	}
	public Subcategory getSubcategory() {
		return subcategory;
	}
	public void setSubcategory(Subcategory subcategory) {
		this.subcategory = subcategory;
	}
	public double getDiscount() {
		return discount;
	}
	public void setDiscount(double discount) {
		this.discount = discount;
	}
	public double getPrice() {
		return price;
	}
	public void setPrice(double price) {
		this.price = price;
	}
	public boolean isIsdeal() {
		return isdeal;
	}
	public void setIsdeal(boolean isdeal) {
		this.isdeal = isdeal;
	}
	public LocalDateTime getCreatedAt() {
		return createdAt;
	}
	public void setCreatedAt(LocalDateTime createdAt) {
		this.createdAt = createdAt;
	}
	public LocalDateTime getUpdatedAt() {
		return updatedAt;
	}
	public void setUpdatedAt(LocalDateTime updatedAt) {
		this.updatedAt = updatedAt;
	}



//===============================================================================



    
    @Column
	private String productname_en;  // English translation
	
	@Column
	private String productname_fr;  // French translation
	
	@Column
	private String description_en;  // English description
	
	@Column
	private String description_fr;  // French description
	// Other fields ...
	
	public String getProductname(Locale locale) {
	    return locale.getLanguage().equals("fr") ? productname_fr : productname_en;
	}
	
	public String getDescription(Locale locale) {
	    return locale.getLanguage().equals("fr") ? description_fr : description_en;
	}
}


//=======================================================================================


